5.0.0 intercepters

jmizell opened the pull request.

jentfoo commented on src/main/java/org/threadly/concurrent/wrapper/intercepter/ExecutorTaskIntercepter.java:
```diff
@@ -10,31 +11,44 @@
- * {@link Executor}.  Please see the javadocs of {@link #wrapTask(Runnable)} for more details 
- * about ways a task can be modified or wrapped.</p>
+ * modified, before being submitted to the pool. This class can be passed a lamba in the 
+ * constructor, or {@link #wrapTask(Runnable)} can be overridden to provide the task which should 
```
> or {@link #wrapTask(Runnable)} can be overridden

I would change this to:

> or you can extend this class and override the function {@link #wrapTask(Runnable)}

People should know that you must extend the class to override wrapTask, but I think it's worth having that key word in the docs.

jentfoo commented on src/main/java/org/threadly/concurrent/wrapper/intercepter/ExecutorTaskIntercepter.java:
```diff
@@ -10,31 +11,44 @@
   protected final Executor parentExecutor;
+  protected final Function<Runnable, Runnable> taskManipulator;
   
   protected ExecutorTaskIntercepter(Executor parentExecutor) {
```
Would you mind adding a javadoc for this constructor saying that you must override the wrapTask function.  I know there was not one previously, but now that there is two constructors I think it would be worth pointing out what you must do.

jentfoo commented on src/main/java/org/threadly/concurrent/wrapper/intercepter/PrioritySchedulerTaskIntercepter.java:
```diff
@@ -10,23 +11,37 @@
  */
-public abstract class PrioritySchedulerTaskIntercepter extends SchedulerServiceTaskIntercepter 
+public class PrioritySchedulerTaskIntercepter extends SchedulerServiceTaskIntercepter 
                                                        implements PrioritySchedulerService {
```
nit, but would you re-align the `implements` when it has to wrap on the next line.  The `i` of implements should be under the `e` of extends.

jentfoo commented:
My comments apply to a couple of the classes, but otherwise LGTM

jmizell committed:
Allow interceptors to accept an lambda

Changed abstract interceptor class to allow submission of a lambda as task modifier.
Classes can still be extended and wrapTask overiden and to handle task interception.

jmizell commented:
Closing, will reopen on threadly project
